### DISCUSSION TITLE
What Video Player works best with Astroid?

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#712] [Q&A] [answered]
Hello Astroid Community

I am looking at hosting videos on my site Joomla/Astroid site hosted at Rochen.
This would at least initially be for demo purposes.
Where people could login and view demos of my work.
No linking to YouTube & Co.

I am not sure exactly how I will end up implement this. But it appears that I need to install some kind of Video Player at the Joomla end to get any of this up and running.

Player options appear to be as follows…

https://allvideoshare.mrvinoth.com/
https://web-dorado.com/products/joomla-player.html
https://yendifplayer.com/

Does any have any tips what would work best with Astroid eg Joomla/Astroid Blog Articles or new Astroid Video Widget?

thanks

Paul

**Comment by Chacapamac** [comment]
I Paul, How You doing.
Just did a small research on my side.

**ANNOUNCEMENT**___________________________________
**Lee** ask for donations by “Buy me a coffee”,  if possible to help him for the time he pass on the project
• Discussion —> https://github.com/templaza/astroid-framework/discussions/705#discussioncomment-9878301
• Donation Page —> https://ko-fi.com/astroidframework/goal?g=0
______________________________________________________________________

**VIDEOS HOSTING OPTIONS:**__________________________________________

— **Self-Hosted Videos:** You can upload videos directly to your Joomla site. [Use HTML5-compatible video players to showcase local or server videos
https://www.gavick.com/documentation/joomla/tutorials/joomla-video-template)[2](https://www.gavick.com/documentation/joomla/tutorials/joomla-video-template

— **YouTube or Vimeo Integration:** Embed videos from platforms like YouTube or Vimeo. Joomla extensions can simplify this process.

— **Joomla Extensions:** Consider using extensions like JoomVideos or JoomBlog for video management within your Joomla site
https://blog.fastdot.com/joomla/what-are-the-best-joomla-tutorials-for-video-content-management/




**VIDEO PLAYERS** ________________________________________________ _(Need HTML5 Player)_

**All Video Share by Vinoth Kumar:**

_This extension is a powerful and free video player and video gallery for Joomla. It’s actively maintained and has been downloaded over 100,000 times. [It’s suitable for YouTubers, video bloggers, course creators, podcasters, and anyone using video on their website](https://extensions.joomla.org/category/multimedia/multimedia-players/)[1](https://extensions.joomla.org/category/multimedia/multimedia-players/)._


**JW Player Advanced by Baboon:**

_JW Player is a well-known HTML5 player that offers responsive playback. The JW Player Advanced extension integrates JW Player 8.32 into Joomla. [You can customize parameters and override settings for a tailored video experience](https://extensions.joomla.org/extension/spider-video-player/)[2](https://extensions.joomla.org/extension/spider-video-player/)._


**Yendif Video Share by Yendif Technologies:**

_Yendif Video Share is an HTML5 video player and video gallery plugin. It supports various video formats (MP4, WebM, OGV) and embeddable players like YouTube, Vimeo, and Facebook. [It’s suitable for both single video players and large video-sharing websites](https://extensions.joomla.org/extension/spider-video-player-lite/)[3](https://extensions.joomla.org/extension/spider-video-player-lite/)._


**Videobox by Hitko Development:**

_Videobox supports videos from YouTube, Vimeo, SoundCloud, Twitch streams, and HTML5-supported video and audio formats. [You can insert videos directly or use pop-up players and video galleries](https://forum.joomla.org/viewtopic.php?t=529326)[4](https://forum.joomla.org/viewtopic.php?t=529326)_

**Comment by paurray** [comment]
Hi Patrick,

thanks for your fast response and suggestions.
I need a little time to work through the links.
And sorry question was a little vague.
Was just wondering if there was solution XYZ that multiple people say is an excellent choice!

The more I think about it the more I think that something like this is the way to go:
(Use the Joomla Articles in combination with the Astroid Framework Blog Article Functionality)
https://astroidframe.work/blog/astroid-hands-on

The key for me is getting none You Tube & Vimeo = Self hosted videos working :-)

Yes I saw the donations page and plan on donating.

Still making Astroid Tutorials (slightly slow due to summer here) and even hit 50 subscribers!

Will be posting Deep Dive: Astroid Boilerplate Site 203 Article next week. Video already finished!

hope you are well and will report back

thanks

Paul

**Comment by paurray** [comment, marked as answer]
@Chacapamac 
Ok I have now understood my own question.
Funnily enough your first link sent me down a rabbit hole and reminded me that I already had JCE Editor up and running. Including by the JCE MediaBox. But I now think that a player like this is not even a requirement. Any way I have done testing here:

https://finalbug.tv/de/paul-david-murray-deutschland/test123

<img width="960" alt="AstroidVideoThumb" src="https://github.com/templaza/astroid-framework/assets/9163424/b918a3bf-f144-4728-81ae-3cbb64003cf8">

Case 01: Adding to Blog Options URL for video to Article Options (Astroid) see above = Only works with You Tube or Vimeo

Case 02: Embedding the same video here with help of JCE MediaBox and insert directly into a Joomla Article = JCE MediaBox might not even be a requirement but works smoother?!?

Case 03: (Please See Footer) Embedding the same video into a Astroid Video Widget = Works!!!

In all cases attempting to add a URL of a local video:
https://finalbug.tv/videos/QuickTest.mp4

I guess I need to post again and change this to a feature request. As in:

**Would it be possible to add a video at a location of choice to the Article Type Options in the Joomla Article section?**

Hope this makes sense now?
